File: nodes/string-match/string-match.process.py
import re

async def process(inputs, settings, config, nodeConfig):
  text = str(inputs.get("text", ""))
  pattern = str(inputs.get("pattern", ""))
  
  try:
    flags = 0
    if not settings.get("case_sensitive", True):
      flags |= re.IGNORECASE
      
    if settings.get("global", False):
      matches = list(re.finditer(pattern, text, flags=flags))
      return {
        "matched": len(matches) > 0,
        "matches": [m.group(0) for m in matches],
        "groups": list(matches[0].groups()) if matches else []
      }
    else:
      match = re.search(pattern, text, flags=flags)
      return {
        "matched": match is not None,
        "matches": [match.group(0)] if match else [],
        "groups": list(match.groups()) if match else []
      }
  except:
    return {
      "matched": False,
      "matches": [],
      "groups": []
    } 

File: sdks/nodejs/nodes/string-match/string-match.process.js
export default async ({inputs, settings, config}) => {

  const text = String(inputs.text || "");
  const pattern = String(inputs.pattern || "");
  
  try {
    const flags = (settings.case_sensitive ? "" : "i") + (settings.global ? "g" : "");
    const regex = new RegExp(pattern, flags);
    
    if (settings.global) {
      const matches = Array.from(text.matchAll(regex));
      return {
        matched: matches.length > 0,
        matches: matches.map(m => m[0]),
        groups: matches.length > 0 ? matches[0].slice(1) : []
      };
    } else {
      const match = text.match(regex);
      return {
        matched: match !== null,
        matches: match ? [match[0]] : [],
        groups: match ? match.slice(1) : []
      };
    }
  } catch (e) {
    return {
      matched: false,
      matches: [],
      groups: []
    };
  }
}; 